internal/stacks/stackruntime/internal/stackeval/main.go

```diff
@@ -128,8 +127,9 @@ func NewForPlanning(config *stackconfig.Config, prevState *stackstate.State, opt
 	return &Main{
 		config: config,
 		planning: &mainPlanning{
-			opts:      opts,
-			prevState: prevState,
+			opts:          opts,
+			prevState:     prevState,
+			planTimestamp: opts.PlanTimestamp,
```

Comment:
I don't even think you need this - just read the timestamp directly from the `opts` whenever you would read it from here.